src/schemas/url.ts

```diff
@@ -0,0 +1,10 @@
+import { z } from 'zod'
+import { HOME } from '~/lib/routes'
+import { isRelativeUrl } from '~/utils/url'
+
+export const callbackUrlSchema = z
+  .string()
+  .optional()
+  .default(HOME)
+  .refine((url) => url && isRelativeUrl(url))
```

Comment:
nit: looking at this again, probably don't need the truthy check with the addition of the `default` clause